tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql

```diff
@@ -43,4 +43,22 @@ select count(null) from numbers(1000);
 
 SELECT max(number) FROM numbers_mt (10) where number > 99999999998;
 SELECT max(number) FROM numbers_mt (10) where number > 2;
-set enable_planner_v2 = 0;
\ No newline at end of file
+
+-- Inner join
+create table t(a int);
+insert into t values(1),(2),(3);
+create table t1(b float);
+insert into t1 values(1.0),(2.0),(3.0);
+create table t2(c uint32 null);
+insert into t2 values(1),(2),(null);
+
+select * from t inner join t1 on cast(t.a as float) = t1.b;
+select * from t inner join t2 on t.a = t2.c;
+select * from t inner join t2 on t.a = t2.c + 1;
```

Comment:
I did not get it, why does this query have empty results?

Reply:
It looks like a bug, this query should output:
```
a c
----
2 1
3 2
```

Just as the following query.

I'll fix it.

Reply:
`select null+1;` will panic in v2:
```
022-05-07T14:07:37.182626Z ERROR common_tracing::panic_hook: panicked at 'called `Option::unwrap()` on a `None` value', query/src/sql/planner/binder/project.rs:44:59 backtrace=Backtrace [{ fn: "common_tracing::panic_hook::set_panic_hook::{{closure}}", file: "./common/tracing/src/panic_hook.rs", line: 25 }, { fn: "std::panicking::rust_panic_with_hook", 
```